prover_rust/src/config.rs

```diff
@@ -0,0 +1,65 @@
+
+use ethers_core::types::BlockNumber;
+use serde::{Deserialize, Serialize};
+// use serde_json::Error;
+use std::error::Error;
+use std::fs::File;
+
+use crate::types::ProofType;
+
+
+#[derive(Debug)]
+#[derive(Serialize, Deserialize)]
+pub struct ProverCoreConfig {
+    pub params_path: String,
+    pub assets_path: String,
+
+    #[serde(default)]
+    pub proof_type: ProofType,
+    #[serde(default)]
+    pub dump_dir: String,
+}
+
+#[derive(Debug)]
+#[derive(Serialize, Deserialize)]
+pub struct CoordinatorConfig{
+    pub base_url: String,
+    pub retry_count: u16,
+    pub retry_wait_time_sec: u32,
+    pub connection_timeout_sec: u32,
+}
+
+#[derive(Debug)]
+#[derive(Serialize, Deserialize)]
+pub struct L2GethConfig{
+    pub endpoint: String,
+    pub confirmations: BlockNumber,
+}
+
+#[derive(Debug)]
+#[derive(Serialize, Deserialize)]
+pub struct Config {
+    pub prover_name: String,
+    pub hard_fork_name: String,
+    pub keystore_path: String,
+    pub keystore_password: String,
+    pub db_path: String,
+    pub core: ProverCoreConfig,
+    pub coordinator: CoordinatorConfig,
+    pub l2geth: L2GethConfig,
```

Comment:
`L2GethConfig` should be optional?

Reply:
agree. batch prover does not need this config.

Reply:
this is done